### DISCUSSION TITLE
Filter collection by `IN` operator on `id` field

### DISCUSSION THREAD

**Discussion by lightningspirit** [pocketbase/pocketbase#4463] [Q&A] [answered]
How can I get a list from a collection `books` by their IDs?
As per [this topic](https://github.com/pocketbase/pocketbase/discussions/434) the `IN` operator wasn't supported at that time. Is it still the case?

Does the following example with an array of IDs work?

```ts
const ids = ["id-1", "id-42"]

const result = await pb.collection("books").getList(1, 20, {
    filter: pb.filter("id ?= {:ids}", { ids })
})
```
Or should I go with a loop in the filter? If that's the case, how to use it properly, having escaping variables in consideration?
Thank you

**Comment by ganigeorgiev** [comment, marked as answer]
The linked discussion is for a different use case.

If you want to fetch the books matching the ids in `ids` using named parameters you can simply combine them with `||`. For example:

```js
const ids = ["id-1", "id-42"]

const records = await pb.collection("books").getFullList({
    // id = "id-1" || id = "id-42"
    filter: ids.map((id) => pb.filter("id = {:id}", { id })).join("||")
})
```

**Reply by lightningspirit** [reply]
Thank you @ganigeorgiev. That's exactly what I was looking for.
